tests/unit/test_rest_streaming_async.py

```diff
@@ -0,0 +1,353 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import logging
+import random
+import time
+from typing import List, AsyncIterator
+import mock
+
+import proto
+import pytest
+
+from google.api_core import rest_streaming_async
+from google.api import http_pb2
+from google.api import httpbody_pb2
+from google.auth.aio.transport import Response
+
+from ..conftest import Composer, Song, EchoResponse, parse_responses
+
+# TODO (ohmayr): check if we need to log.
+__protobuf__ = proto.module(package=__name__)
+SEED = int(time.time())
+logging.info(f"Starting async rest streaming tests with random seed: {SEED}")
+random.seed(SEED)
+
+
+async def mock_async_gen(data, chunk_size=1):
+    for i in range(0, len(data)):
+        chunk = data[i : i + chunk_size]
+        yield chunk.encode("utf-8")
+
+
+class ResponseMock(Response):
+    class _ResponseItr(AsyncIterator[bytes]):
+        def __init__(self, _response_bytes: bytes, random_split=False):
```

Comment:
If we encounter a failure with tests and we have to debug, it's usually a pain to have the test be non-deterministic. I suggest we do one of the following:
* set `random.seed` explicitly in *each* test function. The value can either be hard-coded in (which I would recommend), or it can be varying (by passing in, say, an integer based on time of day, PID, etc.) . In either case, but especially in the latter case, make sure the logs and/or errors print out what the value of the seed turned out to be, so that if we encounter the same failure, we can make sure the test function sets the same seed  so we can replicate it.

or

* pass in the seed to this `__init__`. Same comments above apply for setting in at the calling site.


I realize this was happening in the sync version as well, so it's probably best if you add a TODO in both places and we address this in a follow-up PR. WDYT?

Reply:
I think it's best that we address this for both sync and async together and can do this as a follow up. I've filed https://github.com/googleapis/python-api-core/issues/689 to track this.

Reply:
This has been addressed.